presto-main/src/main/java/com/facebook/presto/operator/PagesRTreeIndex.java

```diff
@@ -116,28 +118,29 @@ private static Envelope getEnvelope(OGCGeometry ogcGeometry)
             return EMPTY_ADDRESSES;
         }
 
-        LongArrayList matchingAddresses = new LongArrayList();
+        IntArrayList matchingPositions = new IntArrayList();
 
         Envelope envelope = getEnvelope(probeGeometry);
         rtree.query(envelope, item -> {
-            GeometryWithAddress geometryWithAddress = (GeometryWithAddress) item;
-            if (spatialRelationshipTest.test(geometryWithAddress.ogcGeometry, probeGeometry)) {
-                matchingAddresses.add(geometryWithAddress.address);
+            GeometryWithPosition geometryWithPosition = (GeometryWithPosition) item;
+            if (spatialRelationshipTest.test(geometryWithPosition.ogcGeometry, probeGeometry)) {
+                matchingPositions.add(geometryWithPosition.position);
             }
         });
 
-        return matchingAddresses.toLongArray(null);
+        return matchingPositions.toIntArray(null);
     }
 
     @Override
-    public boolean isJoinAddressEligible(long joinAddress, int probePosition, Page probe)
+    public boolean isJoinPositionEligible(int joinPosition, int probePosition, Page probe)
     {
-        return filterFunction == null || filterFunction.filter(toIntExact(joinAddress), probePosition, probe);
+        return filterFunction == null || filterFunction.filter(joinPosition, probePosition, probe);
```

Comment:
why `joinPosition` and not `addresses.getLong(joinPosition)` ? I think this is _the_ place which you're fixing, right?

If i understand this correctly, please add to the commit message:
- what was wrong ("FilterFunction was incorrectly invoked")
- something about the renames, that they are many renames, but little logic changed

I wonder if this could be made more visible but somehow splitting the commits.

Also, maybe the `com.facebook.presto.operator.JoinFilterFunction#filter`'s `leftAddress` argument should be renamed to eg `leftAddressIndex`?

Reply:
Also, the commit title line could be improved.

Reply:
@findepi 

> why joinPosition and not addresses.getLong(joinPosition) ?

It turns out that  `leftAddress` in FIlterFunction interface refers not the address, but rather an index in the `addresses` array.

```
    @Override
    public boolean filter(int leftAddress, int rightPosition, Page rightPage)
    {
        long pageAddress = addresses.getLong(leftAddress);
        int blockIndex = decodeSliceIndex(pageAddress);
        int blockPosition = decodePosition(pageAddress);

        return filterFunction.filter(blockPosition, getLeftBlocks(blockIndex), rightPosition, rightPage.getBlocks());
    }
```

Reply:
Yea, i figured that out, but wanted to make sure with you that i understand correctly. Thanks.